container_api/src/interface.rs

```diff
@@ -156,10 +189,158 @@ impl ContainerApiBuilder {
             .insert(instance_name.clone(), instance.clone());
         self
     }
+
+    fn unwrap_params_map(params: Params) -> Result<Map<String, Value>, jsonrpc_core::Error> {
+        match params {
+            Params::Map(map) => Ok(map),
+            _ => Err(jsonrpc_core::Error::invalid_params("expected params map")),
+        }
+    }
+
+    /// To be consistent with holochain function calls the expected output of the admin calls
+    /// is stringified JSON. This performs that stringification and then wraps it in the format expected
+    /// by the jsonrpc add_method closure
+    fn format_response(response: &serde_json::Value) -> jsonrpc_core::Result<serde_json::Value> {
+        Ok(Value::String(serde_json::to_string(response).map_err(
+            |e| jsonrpc_core::Error::invalid_params(e.to_string()),
+        )?))
+    }
+
+    fn get_as_string<T: Into<String>>(
+        key: T,
+        params_map: &Map<String, Value>,
+    ) -> Result<String, jsonrpc_core::Error> {
+        let key = key.into();
+        Ok(params_map
+            .get(&key)
+            .ok_or(jsonrpc_core::Error::invalid_params(format!(
+                "`{}` param not provided",
+                &key
+            )))?
+            .as_str()
+            .ok_or(jsonrpc_core::Error::invalid_params(format!(
+                "`{}` is not a valid json string",
+                &key
+            )))?
+            .to_string())
+    }
+
+    pub fn with_admin_dna_functions(mut self) -> Self {
+        self.io
+            .add_method("admin/dna/install_from_file", move |params| {
+                let params_map = Self::unwrap_params_map(params)?;
+                let id = Self::get_as_string("id", &params_map)?;
+                let path = Self::get_as_string("path", &params_map)?;
+                container_call!(|c| c.install_dna_from_file(PathBuf::from(path), id.to_string()))?;
+                Self::format_response(&json!({"success": true}))
```

Comment:
This produces a result as expected by hc-web-client/hc-redux-middleware and consistent with the other container functions.